sharktank/sharktank/ops/quantized_impls.py

```diff
@@ -52,3 +129,66 @@ def operation_transform(globals: dict[str, Tensor]) -> dict[str, Tensor]:
     return tensor.transform_subtensors(
         operation_transform, copy_external_tensor_trait=False
     )
+
+
+@extract_slice.override(PlanarQuantizedTensor)
+@quantized_tensor_layout_of_type(tensor=BlockScaledFp4Layout)
+def extract_slice_BlockScaledFp4Layout(tensor: PlanarQuantizedTensor, key: Slice):
+    layout: BlockScaledFp4Layout = tensor.layout
+    slice_ = canonicalize_slice_descriptor(squeeze_slice(key), tensor.shape)
+    assert all(
```

Comment:
Using `torch._check` might be necessary for compatibility with torch dynamo, but I haven't thought about how it interacts with the trampoline to sharding.

Reply:
What do you mean? The dispatch mechanism is like for any other tensor type.